prime-router/src/main/kotlin/Element.kt

```diff
@@ -597,11 +628,32 @@ data class Element(
                 } catch (e: DateTimeParseException) {
                     null
                 } ?: try {
-                    // this is a saving throw
+                    // this is a saving throw. the variable date time pattern gives us four different
+                    // types of date patterns to try against, the very last one has nested optional
+                    // portions, which means it can accept a wider array of data, including just date without time
+                    // which we can then coerce to a date time value
                     val optionalDateTime = variableDateTimePattern
                     val df = DateTimeFormatter.ofPattern(optionalDateTime)
-                    val ta = df.parseBest(formattedValue, OffsetDateTime::from, LocalDateTime::from, Instant::from)
-                    LocalDateTime.from(ta).atZone(ZoneOffset.UTC).toOffsetDateTime()
+                    // parseBest makes an attempt to take formatter with a variable pattern and will then
+                    // pick the best return type from the options we pass in, cast as a TemporalAccessor
+                    // which is a wrapper around all the other potential types.
+                    val ta = df.parseBest(
+                        formattedValue,
+                        OffsetDateTime::from,
+                        LocalDateTime::from,
+                        Instant::from,
+                        LocalDate::from
+                    )
+                    // if the TA is a local date time, parse as such and convert to offset
+                    // otherwise, if it's a date, parse just the date type and then upsize to date time
+                    // by assuming start of day. If we aren't given data with an actual time precision
+                    // then pushing it to the start of the day *should* be okay
+                    val parsedValue = if (ta is LocalDateTime) {
+                        LocalDateTime.from(ta).atZone(ZoneId.of(USTimeZone.CENTRAL.zoneId)).toOffsetDateTime()
+                    } else {
+                        LocalDate.from(ta).atStartOfDay(ZoneId.of(USTimeZone.CENTRAL.zoneId)).toOffsetDateTime()
+                    }
+                    parsedValue
                 } catch (e: DateTimeParseException) {
```

Comment:
Also catch DateTimeException here?

Reply:
@clediggins-usds good catch. I’ll add that too.